backend/api/quivr_api/modules/rag_service/rag_service.py

```diff
@@ -263,26 +320,33 @@ async def generate_answer_stream(
             streamed_chat_history.metadata["snippet_emoji"] = (
                 self.brain.snippet_emoji if self.brain else None
             )
-        sources_urls = await generate_source(
-            knowledge_service=self.knowledge_service,
-            brain_id=self.brain.brain_id,
-            source_documents=response.metadata.sources,
-            citations=(
-                streamed_chat_history.metadata["citations"]
-                if streamed_chat_history.metadata
-                else None
-            ),
+
+        sources_urls = (
+            await generate_source(
+                knowledge_service=self.knowledge_service,
+                brain_id=self.brain.brain_id,
+                source_documents=response.metadata.sources,
+                citations=(
+                    streamed_chat_history.metadata["citations"]
+                    if streamed_chat_history.metadata
+                    else None
+                ),
+            )
+            if self.knowledge_service
+            else []
         )
+
         if streamed_chat_history.metadata:
             streamed_chat_history.metadata["sources"] = sources_urls
 
-        self.save_answer(
-            question,
-            ParsedRAGResponse(
-                answer=full_answer,
-                metadata=RAGResponseMetadata.model_validate(
-                    streamed_chat_history.metadata
+        if self.brain_service:
```

Comment:
`brain_service` should probably be required.

Reply:
Not required in chat-with-llm mode... I agree, it is suboptimal, and we should have a BrainService that can also include chat-with-llm, but for now I think that this is fine